custom_build.sh

```diff
@@ -0,0 +1,191 @@
+#!/bin/bash
+
+set -e
+TAG=${1}
+readonly GEOSERVER_VERSION=${2}
+readonly GEOSERVER_MASTER_VERSION=${3}
+readonly GEOSERVER_DATA_DIR_RELEASE=${4}
+readonly PULL=${5}
+readonly ALL_PARAMETERS=$*
+readonly BASE_BUILD_URL="https://build.geoserver.org/geoserver"
+readonly BASE_BUILD_URL_STABLE="https://netcologne.dl.sourceforge.net/project/geoserver/GeoServer"
+readonly EXTRA_FONTS_URL="https://www.dropbox.com/s/hs5743lwf1rktws/fonts.tar.gz?dl=1"
+readonly MARLIN_VERSION=0.9.2
+readonly ARTIFACT_DIRECTORY=./resources
+readonly GEOSERVER_ARTIFACT_DIRECTORY=${ARTIFACT_DIRECTORY}/geoserver/
+readonly DATADIR_ARTIFACT_DIRECTORY=${ARTIFACT_DIRECTORY}/geoserver-datadir/
+readonly PLUGIN_ARTIFACT_DIRECTORY=${ARTIFACT_DIRECTORY}/geoserver-plugins
+readonly FONTS_ARTIFACT_DIRECTORY=${ARTIFACT_DIRECTORY}/fonts/
+readonly MARLIN_ARTIFACT_DIRECTORY=${ARTIFACT_DIRECTORY}/marlin/
+
+function help(){
+	if [ "$#" -ne 6 ] ; then
+		echo "Usage: $0 [docker image tag] [geoserver version] [geoserver master version] [datadir| nodatadir] [pull|no pull];"
+		echo "";
+		echo "[docker image tag] :          the tag to be used for the docker iamge ";
+		echo "[geoserver version] :         the release version of geoserver to be used; you can set it to master if you want the last release";
+		echo "[geoserver master version] :  if you use the master version for geoserver you need to set it to the numerical value for the next release;"
+		echo "                              if you use a released version you need to put it to the release number";
+		echo "[datadir| nodatadir]:         if this parameter is equal to nodatadir the datadir is not burned in the docker images ";
+		echo "[pull|no pull]:               docker build use always a remote image or a local image";
+		exit 1;
+	fi
+}
+
+function clean_up_directory() {
+	rm -rf ${1}/*
+}
+
+function download_from_url_to_a_filepath {
+	URL=${1}
+	FILE_PATH=${2}
+	FILE_DOWNLOADED=$(basename "${FILE_PATH}" )
+	if [ -f "${FILE_PATH}" ]; then
+		rm -f "${FILE_PATH}"
+	fi
+	if [ ! -f "${FILE_PATH}" ]; then
+		curl -L "${URL}" --output "${FILE_PATH}"
```

Comment:
This curl download was not working in my case (for plugin downloads). But i could verify that the generated download URLs for the plugins were correct and working in the browser. So i think this could also be related to the way sourceforge is redirecting here...

Reply:
I confirm I have similar behaviour where the final war.zip for geoserver is instead an HTMLpage @lpasquali
```
/output/webapp# head geoserver-2.18.2-war.zip
<!doctype html>
<!-- Server: sfs-consume-6 -->
<html class="no-js" lang="en">
    <head>
        <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
        <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=no" />


        <script>
    var __gdpr = true;
...
```

Reply:
@buehner how was the url this time?